### ISSUE TITLE
Breaks on node 11

### ISSUE THREAD

**Issue by hueniverse** [hapijs/nes#257]
I removed node 11 from travis until this is resolved.

@cjihrig can you also take a look?

**Comment by cjihrig** [comment]
This appears to be a timers bug in Node core. On my machine (macOS), Node 11.1.0 and latest Node master branch (Node 10 is not impacted), the following command just hangs:

```
$ node node_modules/.bin/lab -i 113
```

The `setTimeout()` inside [this](https://github.com/hapijs/nes/blob/819b6c402ed416de318e46f7845bee78c6b488e2/test/listener.js#L176) call to `Hoek.wait()` never calls its callback.

I've been able to reproduce this without `lab` and `hoek`, but haven't put together a reproduction without hapi+nes.

<details>

```js
'use strict';
const Hapi = require('hapi');
const Nes = require('./');

async function main() {
  const server = Hapi.server();
  await server.register({
    plugin: Nes,
    options: {
      auth: false,
      heartbeat: { interval: 50, timeout: 45 }
    }
  });

  server.route({
    method: 'GET',
    path: '/',
    handler: async () => {
      await wait(110);
      return 'hello';
    }
  });

  await server.start();
  const client = new Nes.Client('http://localhost:' + server.info.port);

  await client.connect();
  await client.request('/');
  client.disconnect();
  await server.stop();
}

setTimeout(() => {}, 1000);
main();


function wait(timeout) {
  return new Promise((resolve) => {
      setTimeout(resolve, timeout);
  });
};
```
</details>

**Comment by cjihrig** [comment]
Opened https://github.com/nodejs/node/issues/24320.

**Comment by cjihrig** [comment]
@hueniverse this should be fixed in Node 11.2.0, which was just released. The nes test suite passes for me locally on 11.2.0.
